### PR TITLE
v7.17.0

### PR DESCRIPTION
**Pull request by arthurbalduini**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,92 @@
...
+- [docs] Match title with blog posts @oliviertassinari
+
+### Core
+
+- [core] Move warning methods to `@mui/x-internals` (#14528) @k-rajat19
+- [core] Sync with core release flow @oliviertassinari
+- [code-infra] Fix charts benchmark workflow (#14573) @JCQuintas
+- [docs-infra] Type interface API pages (#14138) @alexfauquette
+- [infra] Create `ESLint plugins` renovate group (#14574) @LukasTy
+- [infra] Fix renovate ESLInt group config (#14577) @LukasTy
```
Comment (LukasTy) [inline CHANGELOG.md:88]: This is just a fix of the above change. 🙈 
```suggestion
```

File: packages/x-charts-pro/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-charts-pro",
-  "version": "7.0.0-beta.0",
+  "version": "7.0.1-beta.0",
```
Comment (LukasTy) [inline packages/x-charts-pro/package.json:3]:
```suggestion
  "version": "7.0.0-beta.1",
```

**Review by LukasTy — COMMENTED** [review]
At least a few new public APIs were added to Pickers, I'm leaning towards a minor release, what do others think?

@flaviendelangle do you think it's too early to highlight #14514 and #14486?

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,92 @@
...
+
+- [DateRangePicker] Fix `currentMonthCalendarPosition` not scrolling to future sibling (#14442) @GMchris
+
+### Charts
+
+#### `@mui/x-charts@7.16.1`
+
+- [charts] Add `"use client"` directive to every public component and hook (#14578) @flaviendelangle
+- [charts] Allow `onItemClick` on the `Legend` component (#14231) @JCQuintas
+- [charts] Fix `onAxisClick` with layour='horizontal' (#14547) @alexfauquette
```
Comment (alexfauquette) [inline CHANGELOG.md:60]:
```suggestion
- [charts] Fix `onAxisClick` with `layour='horizontal'` (#14547) @alexfauquette
```

**Review by cherniavskii — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,92 @@
...
+
+### Data Grid
+
+#### `@mui/x-data-grid@7.16.1`
+
+- [DataGrid] Add "does not equal" and "does not contain" filter operators (#14489) @KenanYusuf
+- [DataGrid] Add demo to the "Custom columns" page that does not use generator (#13695) @arminmeh
+- [DataGrid] Fix Voice Over reading the column name twice (#14482) @arminmeh
+- [DataGrid] Fix bug in CRUD example (#14513) @michelengelen
+- [DataGrid] Refactor string operator filter functions (#14564) @KenanYusuf
```
Comment (KenanYusuf) [inline CHANGELOG.md:27]:
```suggestion
- [DataGrid] Fix failing jsdom tests caused by :has selectors (#14559) @KenanYusuf
- [DataGrid] Refactor string operator filter functions (#14564) @KenanYusuf
```

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,93 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.16.1
+
+_Sep 12, 2024_
+
+We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:
+
+- 📊 Charts performance improvement
+- 🧑‍💻 New Data Grid [custom columns demo](https://mui.com/x/react-data-grid/custom-columns/#full-example)
```
Comment (michelengelen) [inline CHANGELOG.md:13]: we never used this emoji before ... shouldn't we stick to the ones we use frequently?
Also what should this demonstrate? It's not clear to me.

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,93 @@
...
+
+### Data Grid
+
+#### `@mui/x-data-grid@7.16.1`
+
+- [DataGrid] Add "does not equal" and "does not contain" filter operators (#14489) @KenanYusuf
+- [DataGrid] Add demo to the "Custom columns" page that does not use generator (#13695) @arminmeh
+- [DataGrid] Fix Voice Over reading the column name twice (#14482) @arminmeh
+- [DataGrid] Fix bug in CRUD example (#14513) @michelengelen
+- [DataGrid] Fix failing jsdom tests caused by `:has` selectors (#14559) @KenanYusuf
```
Comment (michelengelen) [inline CHANGELOG.md:28]: nitpick

```suggestion
- [DataGrid] Fix failing jsdom tests caused by `:has()` selectors (#14559) @KenanYusuf
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,93 @@
...
+
+- [DateRangePicker] Fix `currentMonthCalendarPosition` not scrolling to future sibling (#14442) @GMchris
+
+### Charts
+
+#### `@mui/x-charts@7.16.1`
+
+- [charts] Add `"use client"` directive to every public component and hook (#14578) @flaviendelangle
+- [charts] Allow `onItemClick` on the `Legend` component (#14231) @JCQuintas
+- [charts] Fix `onAxisClick` with `layour='horizontal'` (#14547) @alexfauquette
```
Comment (michelengelen) [inline CHANGELOG.md:62]:
```suggestion
- [charts] Fix `onAxisClick` with `layout='horizontal'` (#14547) @alexfauquette
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,93 @@
...
+- [charts] Replace `path` with `circle` for perf improvement (#14518) @alexfauquette
```
Comment (michelengelen) [inline CHANGELOG.md:63]:
```suggestion
- [charts] Replace `path` with `circle` for performance improvement (#14518) @alexfauquette
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,93 @@
...
+
+#### `@mui/x-charts-pro@7.0.1-beta.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-charts@7.16.1`.
+
+### Tree View
+
+#### `@mui/x-tree-view@7.16.1`
+
+- [TreeView] Make `useTreeItem2` stable (#14498) @flaviendelangle
```
Comment (michelengelen) [inline CHANGELOG.md:73]: nitpick:
We should decide how to tag PRs at some point.
the grid PRs are always tagged with the component name and pickers and charts use the generic one.

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,93 @@
...
+
+### Docs
+
+- [docs] Add missing callout on Imperative API tree view sections (#14503) @flaviendelangle
+- [docs] Fix broken redirection to MUI X v5 @oliviertassinari
+- [docs] Fix multiple `console.error` messages on `charts` docs (#14554) @JCQuintas
+- [docs] Fixed typo in Recipes Row Grouping (#14549) @Miodini
```
Comment (michelengelen) [inline CHANGELOG.md:80]:
```suggestion
- [docs] Fixed typo in Recipes/Row Grouping (#14549) @Miodini
```

or 

```suggestion
- [docs] Fixed typo in Row Grouping recipes (#14549) @Miodini
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,93 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.16.1
```
Comment (LukasTy) [inline CHANGELOG.md:6]: IMHO, given at least Tree View and Pickers changes this should be a minor release. 🤔 
```suggestion
## 7.17.0
```

File: CHANGELOG.md (hunk 5 above)
Comment (arthurbalduini) [reply CHANGELOG.md:13]: I think it demonstrates something like an user accessing the new demo, but I think @cherniavskii will know better what he meant

Comment (flaviendelangle) [reply CHANGELOG.md:13]: We never really defined what the emoji meant though :laughing:

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,6 +3,93 @@
...
+
+_Sep 12, 2024_
+
+We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:
+
+- 📊 Charts performance improvement
+- 🧑‍💻 New Data Grid [custom columns demo](https://mui.com/x/react-data-grid/custom-columns/#full-example)
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+- 🌍 Improve Hungarian (hu-HU) locales on the Data Grid
```
Comment (flaviendelangle) [inline CHANGELOG.md:16]:
```suggestion
- 🌍 Improve Hungarian (hu-HU) locale on the Data Grid
```

File: CHANGELOG.md (hunk 9 above)
Comment (flaviendelangle) [reply CHANGELOG.md:73]: The rational being that for the grid and the tree view, each package is basically a giant component with some stuff around it (but everything only lives for the needs of the main component).

But as we gain new packages I wouldn't be against the idea of aligning every package to use the generic `grid` `treeview` etc...

File: CHANGELOG.md [hunk 13]
```diff
@@ -3,6 +3,93 @@
...
+
+### Tree View
+
+#### `@mui/x-tree-view@7.16.1`
+
+- [TreeView] Make `useTreeItem2` stable (#14498) @flaviendelangle
+
+### Docs
+
+- [docs] Add missing callout on Imperative API tree view sections (#14503) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:77]:
```suggestion
- [docs] Add missing callout on "Imperative API" tree view sections (#14503) @flaviendelangle
```

**Review by flaviendelangle — COMMENTED** [review]
Not that small of a release this week :smiley:

**Comment by arthurbalduini** [comment]
> > At least a few new public APIs were added to Pickers, I'm leaning towards a minor release, what do others think?
> 
> I'm fine having those new APIs in the changelog, maybe not highlighted because they are not documented yet (this will come with the read-only recipe I guess). Fully agree that they justify a minor @LukasTy

Since there are other API changes on Charts in Tree View, would everyone agree on adding a generic highlight? Something like:
`- ⛏️ API additions in Time Pickers, Charts and Tree View`

@flaviendelangle @JCQuintas

**Review by JCQuintas — APPROVED** [review]

**Comment by flaviendelangle** [comment]
> Since there are other API changes on Charts in Tree View, would everyone agree on adding a generic highlight? Something like:
> - ⛏️ API additions in Time Pickers, Charts and Tree View

Not sure that really needed
If the API is interesting enough to highlight, we highlight it, if not we don't
The more time flies, the less I am convinced by the "docs improvements" and "bug fixes", to me they feel like we try to justify that we did stuff.

File: CHANGELOG.md (hunk 5 above)
Comment (cherniavskii) [reply CHANGELOG.md:13]: Yeah, I didn't think much about the emoji choice, I just thought it fits here 😅

**Review by michelengelen — APPROVED** [review]

File: CHANGELOG.md [hunk 14]
```diff
@@ -3,6 +3,93 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.17.0
+
+_Sep 12, 2024_
```
Comment (LukasTy) [inline CHANGELOG.md:8]:
```suggestion
_Sep 13, 2024_
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -3,6 +3,93 @@
...
+- 🧑‍💻 New Data Grid [custom columns demo](https://mui.com/x/react-data-grid/custom-columns/#full-example)
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+- 🌍 Improve Hungarian (hu-HU) locale on the Data Grid
+
+<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->
+
+### Data Grid
+
+#### `@mui/x-data-grid@7.16.1`
```
Comment (LukasTy) [inline CHANGELOG.md:22]: We still need to update all changelog versions and packages. 😉

File: packages/x-date-pickers-pro/package.json [hunk 16]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-date-pickers-pro",
-  "version": "7.16.0",
+  "version": "7.0.0-beta.1",
```
Comment (LukasTy) [inline packages/x-date-pickers-pro/package.json:3]: FLAG Hm... Are you sure?

**Pull request merged**